src/widget/woverview.cpp

```diff
@@ -529,7 +529,16 @@ void WOverview::mousePressEvent(QMouseEvent* e) {
             }
             if (pHoveredCue != nullptr) {
                 m_pCueMenuPopup->setTrackAndCue(m_pCurrentTrack, pHoveredCue);
-                m_pCueMenuPopup->popup(e->globalPos());
+
+                // Shift the popup left if it would go off screen shown at the
+                // current cursor position.
+                QPoint popupPoint = e->globalPos();
+                int screenWidth = windowHandle()->screen()->size().width();
+                if (popupPoint.x() + m_pCueMenuPopup->width() > screenWidth) {
+                    popupPoint.setX(screenWidth - m_pCueMenuPopup->width());
```

Comment:
Do we have to take into account that a screen's origin might not be (0, 0)? This information is encoded in the QRect returned by QScreen::geometry().

How about vertical fitting? Calculating how to offset the global position such that the rectangle to be displayed fits onto the screen could become a generic utility function:

```
/// Returns an adjusted upper left point for displaying the popup
/// with the given size on the screen.
QPoint mapPopupToScreen(
    const QScreen& screen,
    const QPoint& popupUpperLeft,
    const QSize& popupSize);
```

I wonder if Qt doesn't provide this function already? This should be very common.

Reply:
Also, what about screen height? Do we want to account for that, too?

Reply:
Yes, "vertical fitting" = screen height. This function should make adjustments in both directions if required.

Reply:
Sorry, I misread your comment. :-) I agree that it would be nice to have such a general utility function that takes care of both.

Reply:
I agree that moving this to a utility function would be better. I don't think any of the current files in src/util would be an appropriate place, so what should the new file be called? src/util/widgetutils?

Reply:
Qt takes care of placing tooltips and context menus appropriately.
@holzhaus I don't remember, what was in the way to make the cueMENUpopup an actual menu?

Reply:
I can't remember. You'd need to make it a `QWidgetAction` subclass, and setting a menu on a QPushButton might have unexpected consequences (they are drawn differently AFAIK). But if you think it's a worthwhile approach, go ahead.

Reply:
I factored it out to a new utility function in f0c33c7.